### DISCUSSION TITLE
HTTP web server should support chunked/streamed/resumable file uploads

### DISCUSSION THREAD

**Discussion by dumblob** [vlang/v#11882] [Ideas]
Currently the web server seems to assume a file will be uploaded as a whole. But that's not necessarily the case and actually for performance reasons (both significantly lower memory usage as well as starvation avoidance thus much higher efficiency of resources utilization) and to allow writing a "pause & resume" uploader supporting multihoming a chunked/streamed file uploads would be necessary.

Some good inspiration might be [tus](https://tus.io/ ) (used by e.g. Vimeo) or https://github.com/sailshq/skipper .

@mcastorina and others what do you think?

**Comment by JalonSolov** [comment]
The net code already supports chunked downloads, just for older systems.  However, chunked is deprecated as of HTTP 2.  I think V should keep support for chunked downloads, but not bother with trying to support chunked uploads.

**Comment by dumblob** [comment]
That's true it's deprecated and I agree it might not be worth it any more on the server side.

What I think though *V* should support is HTTP 1.1 with chunked uploads in a client role. Especially because HTTP 1.1 will stay with us probably for several more decades.

**Comment by lmangani** [comment]
Has anyone implemented any workaround to process chunked requests from clients using the net.http.chunked functions in vweb?

**Reply by dumblob** [reply]
@lmangani I don't know of any. I'd myself be interested in that. Maybe you can ask on Discord? The community is allegedly pretty active there.
